### PR TITLE
solution

### PR DESCRIPTION
**Pull request by RomSmile**

### CODE REVIEW

**Review by Clonvhs — CHANGES_REQUESTED** [review]
Task was made withou IF ELSE

File: src/averageAges.js [hunk 1]
```diff
@@ -14,12 +15,22 @@
...
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = 0) {
+  const arrayOfMen = people.filter((person) => person.sex === 'm');
+
+  function callback(sum, year) {
+    return sum + year.died - year.born;
+  }
+
+  if (century === 0) {
```
Comment (AndriyNikorych) [inline src/averageAges.js:25]: Завдання було, щоб не використовувати if (replace `if ()` statement with &&, || or ?:)

File: src/averageAges.js [hunk 2]
```diff
@@ -14,12 +15,22 @@
...
+    return arrayOfMen.reduce(callback, 0) / arrayOfMen.length;
+  }
+
+  const arrayOfMenInThisCentury
+    = arrayOfMen.filter((year) => Math.ceil(year.died / 100) === century);
+
+  // eslint-disable-next-line max-len
```
Comment (danheim) [inline src/averageAges.js:32]: you are not allowed to dislable eslint

File: src/averageAges.js (hunk 1 above)
Comment (danheim) [reply src/averageAges.js:25]: avoid if

File: src/averageAges.js [hunk 3]
```diff
@@ -14,12 +15,22 @@
...
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = 0) {
+  const arrayOfMen = people.filter((person) => person.sex === 'm');
```
Comment (danheim) [inline src/averageAges.js:19]: remove `arrayOf`, just `men`

**Review by danheim — CHANGES_REQUESTED** [review]
You are not allowed to disable eslint.

**Review by danheim — APPROVED** [review]

**Pull request closed without merging**